x-pack/plugins/security_solution/server/lib/detection_engine/signals/build_rule.ts

```diff
@@ -103,6 +103,12 @@ export const buildRule = ({
     created_by: createdBy,
     updated_by: updatedBy,
     threat: ruleParams.threat ?? [],
+    threat_mapping: ruleParams.threatMapping ?? [],
```

Comment:
wouldn't we expect these to be left off of `signal.rule` when they are undefined for other rule types rather than defaulted to `[]`?

Reply:
I followed the other examples, above this one. We can change this behavior with a follow up if we don't want empty arrays.

Reply:
Follow up:
https://github.com/elastic/kibana/pull/93063